third_party/nvfuser/csrc/scheduler/reduction_utils.h

```diff
@@ -29,6 +29,38 @@ TORCH_CUDA_CU_API void multiReductionInliner(
     std::vector<std::pair<TensorView*, TensorView*>> cached_outputs,
     std::vector<TensorView*> dummy_outputs = {});
 
+// Propagate transformations with inernal cutoff boundary at boundaryNodesSet
+// in P2C forward propagate, disable propagation to TensorView in
+// boundaryNodesSet in C2P backward propagate, disable propagation from
+// TensorView in boundaryNodesSet
+TORCH_CUDA_CU_API void propagateTransformation(
+    TensorView* reference_tv,
+    const std::unordered_set<TensorView*>& boundaryNodesSet);
+
+// Propagate RFactor from first reduction TensorView to others
+TORCH_CUDA_CU_API void propagateRFactor(
+    TensorView* reference_tv,
+    std::vector<TensorView*> reduction_tvs);
+
+// Propagate Parallelization from reference TensorView to others
```

Comment:
Please add more info about the function parameters. The comment isn't clear what the "others" mean. I suppose it means all the other tensors in the given fusion, right?  What do `unroll` and `vectorize` mean?

Reply:
added.